k8s/client.go

```diff
@@ -808,6 +808,29 @@ func getCiliumVersionFromImage(image string) (string, error) {
 	return version, nil
 }
 
+// GetCiliumVersion returns a semver.Version representing the version of cilium
+// running in the cilium-agent pod
+func (c *Client) GetCiliumVersion(ctx context.Context, p *corev1.Pod) (*semver.Version, error) {
```

Comment:
Could you please split the changes factoring out `GetCiliumVersion` into a separate commit? This will improve bisectability of the commit history.